src/TableauConnector.js

```diff
@@ -304,17 +308,91 @@ export default class TableauConnector {
       }
 
       table.appendRows(tableData)
-      callback()
-    }).catch(function (error) {
-      tableau.log(error)
-      tableau.log('There was an error retrieving the data')
+      dataCallback()
+      utils.log('SUCCESS: Data')
+    }).catch(error => {
       tableau.abortWithError(error)
+      utils.log(`FAILURE: Data (${error})`)
+    })
+  }
+
+  redirect401 (req) {
```

Comment:
You can accomplish this with an interceptor, https://github.com/axios/axios#interceptors
That does assume you always want to perform that action on a 401.

Reply:
This form of error handling only applies in some cases. `TableauConnector.js` knows when to use it and it's not aware of `axios`.